grpc/tls/provider/meshca/v1/config.proto

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package grpc.tls.provider.meshca.v1;
+
+import "envoy/config/core/v3/config_source.proto";
+import "google/protobuf/duration.proto";
+
+option java_outer_classname = "MeshCaConfigProto";
+option java_multiple_files = true;
+option java_package = "io.grpc.tls.provider.meshca.v1";
+
+// GoogleMeshCaConfig contains all configuration parameters required by the
+// MeshCA CertificateProvider plugin implementation.
+message GoogleMeshCaConfig {
+  // Type of key to be embedded in CSRs sent to the MeshCA.
+  enum KeyType {
+    KEY_TYPE_UNKNOWN = 0;
+    KEY_TYPE_RSA = 1;
+  }
+
+  // GoogleMeshCA server endpoint to get CSRs signed via the *CreateCertificate*
+  // unary call. This must have :ref:`api_type
+  // <envoy_api_field_config.core.v3.ApiConfigSource.api_type>` :ref:`GRPC
+  // <envoy_api_enum_value_config.core.v3.ApiConfigSource.ApiType.GRPC>`.
+  // STS based call credentials need to be supplied in :ref:`call_credentials
+  // <envoy_api_field_config.core.v3.GrpcService.GoogleGrpc.call_credentials>`.
+  // If :ref:`timeout envoy_api_field_config.core.v3.GrpcService.timeout` is
+  // left unspecified, a default value of 10s will be used.
+  envoy.config.core.v3.ApiConfigSource server = 1;
```

Comment:
Why is this included? It sounded like this plugin was only going to work with GCP, so what's the point in passing this to the client? Googleapis client libraries generally have the service name hard-coded in them. How is this different?

Now, if we do remove this, maybe we need a new field for `timeout`, or something like that. But I'm trying to understand the design rational for including it here.

Reply:
We're going to want to use the same config proto in xDS for both gRPC and Envoy clients, so that control planes don't have to send different configs to different clients for the same functionality.  Envoy is going to want to encode this information as an `ApiConfigSource`, so that it can use its existing machinery to manage the connection to the MeshCA server.